hibernate-core/src/main/java/org/hibernate/loader/plan/exec/internal/LoadQueryJoinAndFetchProcessor.java

```diff
@@ -222,16 +222,43 @@ private AssociationType getJoinedAssociationTypeOrNull(Join join) {
 				: null;
 	}
 
-	private String resolveAdditionalJoinCondition(String rhsTableAlias, String withClause, Joinable joinable, AssociationType associationType) {
+	private String resolveAdditionalJoinCondition(
+			Join join,
+			String rhsTableAlias,
+			String withClause,
+			Joinable joinable,
+			AssociationType associationType) {
 		// turns out that the call to AssociationType#getOnCondition in the initial code really just translates to
 		// calls to the Joinable.filterFragment() method where the Joinable is either the entity or
 		// collection persister
-		final String filter = associationType!=null?
-				associationType.getOnCondition( rhsTableAlias, factory, queryInfluencers.getEnabledFilters() ):
-				joinable.filterFragment(
+
+		final String filter;
+		// If the left hand side
+		if ( associationType.isEntityType() &&
```

Comment:
You will have a NPE here if `associationType` is null whereas before the code would support a null `associationType`.

Reply:
Yes, you're right. I'll fix that.